hbase-balancer/src/main/java/org/apache/hadoop/hbase/master/balancer/StochasticLoadBalancer.java

```diff
@@ -372,29 +389,37 @@ boolean needsBalance(TableName tableName, BalancerClusterState cluster) {
       return true;
     }
 
-    if (sloppyRegionServerExist(cs)) {
+    if (!balancerConditionals.shouldSkipSloppyServerEvaluation() && sloppyRegionServerExist(cs)) {
```

Comment:
If we should figure out how to have these more nicely coexist, then I'm open to that feedback. In my opinion, based on my experience with the balancer, this is a fine tradeoff